### PR TITLE
#4061 extended secretUtils

### PR DESCRIPTION
**Pull request by bacherfl**
Parat of https://github.com/keptn/keptn/issues/4061
Signed-off-by: Florian Bacher <florian.bacher@dynatrace.com>

### CODE REVIEW

File: pkg/api/utils/secretUtils.go [hunk 1]
```diff
@@ -85,24 +88,70 @@ func (s *SecretHandler) getHTTPClient() *http.Client {
...
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errObj := &models.Error{}
+		if err := json.Unmarshal(body, errObj); err != nil {
```
Comment (warber) [inline pkg/api/utils/secretUtils.go:147]: we should re-add the methods for (de)serializing to the structs itself. I've already done this for the e.g. here: https://github.com/keptn/go-utils/blob/21f45c833164c75788f2bd93e02ae9acd54452cb/pkg/api/models/uniform.go#L67

but lets do this in a separate PR for all models?

**Review by warber — APPROVED** [review]

File: pkg/api/utils/secretUtils.go (hunk 1 above)
Comment (bacherfl) [reply pkg/api/utils/secretUtils.go:147]: agreed, let's add that afterwards
